dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go

```diff
@@ -302,38 +314,48 @@ func generateTarball(skipUI, skipHelm bool) error {
 	run(fmt.Sprintf("removing any existing %v", dstPath), "rm", "-rf", dstPath)
 	fmt.Printf("Creating %s:\n", tarball)
 
-	for _, dir := range []string{
-		"assembly", "client", "logs", "integration/fuse", "integration/kubernetes", "logs/user",
-	} {
-		mkdir(filepath.Join(dstPath, dir))
+	toCreateDirs := []string{"logs", "lib", "bin"}
+	if !opts.Fuse {
+		toCreateDirs = append(toCreateDirs, "assembly", "client", "integration/fuse", "integration/kubernetes", "logs/user")
 	}
-
-	if err := os.Chmod(filepath.Join(dstPath, "logs/user"), 0777); err != nil {
-		return err
+	for _, dir := range toCreateDirs {
+		mkdir(filepath.Join(dstPath, dir))
 	}
 
-	run("adding Alluxio client assembly jar", "mv", fmt.Sprintf("assembly/client/target/alluxio-assembly-client-%v-jar-with-dependencies.jar", version), filepath.Join(dstPath, "assembly", fmt.Sprintf("alluxio-client-%v.jar", version)))
-	run("adding Alluxio server assembly jar", "mv", fmt.Sprintf("assembly/server/target/alluxio-assembly-server-%v-jar-with-dependencies.jar", version), filepath.Join(dstPath, "assembly", fmt.Sprintf("alluxio-server-%v.jar", version)))
-	run("adding Alluxio FUSE jar", "mv", fmt.Sprintf("integration/fuse/target/alluxio-integration-fuse-%v-jar-with-dependencies.jar", version), filepath.Join(dstPath, "integration", "fuse", fmt.Sprintf("alluxio-fuse-%v.jar", version)))
-	// Generate Helm templates in the dstPath
-	run("adding Helm chart", "cp", "-r", filepath.Join(srcPath, "integration/kubernetes/helm-chart"), filepath.Join(dstPath, "integration/kubernetes/helm-chart"))
-	if !skipHelm {
-		chdir(filepath.Join(dstPath, "integration/kubernetes/helm-chart/alluxio/"))
-		run("generate Helm templates", "bash", "helm-generate.sh", "all")
-		chdir(srcPath)
-	}
+	if opts.Fuse {
```

Comment:
i wonder how much of the tarball generation codepath is share with fuse vs not. is it worth combining into one codepath...?

Reply:
Sort of hard to decide whether to share or not, share code is non-trivial and non-shared code is also non-trivial